### ISSUE TITLE
Can't change background of canvas

### ISSUE THREAD

**Issue by jauggy** [storybookjs/storybook#12305]
**Describe the bug**
I can't change the background of the canvas. Also looking at the tutorials the canvas should have white background but for some reason mine is grey.

**To Reproduce**
Clone repo at: https://github.com/jauggy/storybook-background-issues

```npm i```

```npm run storybook```

Notice the Canvas background is gray. Also when clicking the background addons you can't change it.

**Expected behavior**
White background on canvas and ability to change the canvas background via the background addon.

**Screenshots**
![image](https://user-images.githubusercontent.com/1305569/91630715-14a92c00-ea17-11ea-9873-5e84b66a81bd.png)

**Code snippets**
If applicable, add code samples to help explain your problem.

**System:**
```
Environment Info:

  System:
    OS: Windows 10 10.0.19041
    CPU: (4) x64 Intel(R) Core(TM) i5-2500 CPU @ 3.30GHz
  Binaries:
    Node: 12.18.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.5 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 84.0.4147.135
    Edge: Spartan (44.19041.423.0), Chromium (85.0.564.41)
  npmPackages:
    @storybook/addon-essentials: ^6.0.20 => 6.0.20
    @storybook/react: ^6.0.20 => 6.0.20
```

**Additional context**

**Comment by ajkl2533** [comment]
I had the same issue which was caused by setting `background-color` on `body` element.

**Comment by monolithed** [comment]
I have the same problem. It seems the plugin does not work.
My temporary solution is:

```js
// preview.js

import React from 'react';
import {themes} from '@storybook/theming';

const theme = process.env.STORYBOOK_THEME || themes.light;

const Theme = {
    LIGHT: 'Светлая тема',
    DARK: 'Темная тема',
};

const globalTypes = {
    theme: {
        name: 'Theme',
        description: 'Изменение темы',
        defaultValue: Theme.LIGHT,
        toolbar: {
            icon: 'circlehollow',
            items: [
                Theme.LIGHT,
                Theme.DARK
            ]
        }
    }
};

const parameters = {
    docs: {
        theme
    },

    backgrounds: {
        default: Theme.LIGHT,
        values: [
            {
                name: Theme.LIGHT,
                value: '#fff'
            },
            {
                name: Theme.DARK,
                value: '#2f2f2f'
            }
        ]
    }
};

const getThemeColor = (theme) => {
    const {value} = parameters.backgrounds.values.find(({name}) => name === theme);

    return value;
};

const withThemeProvider = (Story,context) => {
    const {theme} = context.globals;
    const backgroundColor = getThemeColor(theme);

    document.body.style.backgroundColor = backgroundColor;

    return (
        <div style={{backgroundColor}}>
            <Story {...context} />
        </div>
    );
};

const decorators = [withThemeProvider];

export {
    globalTypes,
    decorators,
    parameters
};
```

**Comment by shilman** [comment]
@yannbf can you give a hand?

**Comment by yannbf** [comment]
Hey @jauggy the reason the addon does not work is that the lib you're using, CoreUI, sets a default background color at the body level. 
This is the line of code that gives the side effect: 
```
@import "~@coreui/coreui/scss/coreui"
```

In storybook 6.0, the background is set at the iframe level, which is less specific than the body. In Storybook 6.1 this problem is fixed. This is your storybook running with `^6.1.0-alpha.23`:
![Kapture 2020-10-15 at 12 00 01](https://user-images.githubusercontent.com/1671563/96108630-199a4000-0ede-11eb-8e83-54003aa14735.gif)


Also, Storybook 6 supports typescript out of the box! So you can remove these lines from your main.js:
```diff
webpackFinal: async (config) => {
    config.module.rules.push({
      test: /\.scss$/,
      use: ["style-loader", "css-loader", "sass-loader"],
      include: path.resolve(__dirname, "../"),
    });

-    config.module.rules.push({
-      test: /\.(ts|tsx)$/,
-      loader: require.resolve("babel-loader"),
-      options: {
-        presets: [["react-app", { flow: false, typescript: true }]],
-      },
-    });
-    config.resolve.extensions.push(".ts", ".tsx");

    return config;
},
```

You can also use main.ts and preview.(ts|tsx)!

**Comment by monolithed** [comment]
@yannbf thanks a lot. It works!

**Comment by monolithed** [comment]
<img width="876" alt="Снимок экрана 2020-10-15 в 19 54 35" src="https://user-images.githubusercontent.com/803674/96161718-4a9b6480-0f20-11eb-8521-bafb87e93b43.png">
<img width="879" alt="Снимок экрана 2020-10-15 в 19 54 21" src="https://user-images.githubusercontent.com/803674/96161722-4bcc9180-0f20-11eb-8ed4-d20331e29fb1.png">
<img width="1943" alt="Снимок экрана 2020-10-15 в 19 53 33" src="https://user-images.githubusercontent.com/803674/96161770-5f77f800-0f20-11eb-9b5f-277f2ce21a8e.png">

**upd** @yannbf , I ran it — there seems to be a problem with docs.

**Comment by yannbf** [comment]
@monolithed good that you had your problem solved! From your screenshots I can't really get much info. If you have a reproduction repo it would be easier to help!

**Comment by monolithed** [comment]
@yannbf The problem is that I cannot change a background color in the Doc tab (

**Comment by yannbf** [comment]
@monolithed thanks! That was clear from your screenshots, but without a reproduction repo I can't infer anything more than that.

**Comment by petomi** [comment]
I had the same issue in 6.0.5 (docs Canvas elements not using the same background-color that they do in the sandbox mode) but this issue appears to be solved by updating to Storybook 6.1.1.

**Comment by monolithed** [comment]
The problem is fixed for me by installing 6.1.1

**Comment by shilman** [comment]
Thanks for the update @monolithed. Closing!
